src/dev-app/dev-app/dev-app-theme.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Inject, Injectable} from '@angular/core';
+import {OverlayContainer} from '@angular/cdk/overlay';
+import {DOCUMENT} from '@angular/common';
+
+const isDarkThemeKey = 'isDarkTheme';
+
+@Injectable({providedIn: 'root'})
+export class DevAppTheme {
+  readonly darkThemeClass = 'demo-unicorn-dark-theme';
+  private _isDark = false;
+
+  constructor(private _overlayContainer: OverlayContainer,
+              @Inject(DOCUMENT) private _document: Document) {
+    let isDark: string | null;
+    try {
+      isDark = localStorage.getItem(isDarkThemeKey);
+      if (isDark != null) {
+        this._isDark = isDark === 'true';
+        this._isDark ? this.applyDarkTheme() : this.applyLightTheme();
```

Comment:
We avoid calling the setter and apply the themes directly here to avoid writing the same value we just read back to `localStorage`.

Reply:
Should that be in a comment?